app/components/ui/Navigation.tsx

```diff
@@ -1,195 +1,312 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import styles from "components/ui/Navigation.module.scss";
+import mobileNavigationStyles from "components/ui/MobileNavigation.module.scss";
+import desktopNavigationStyles from "components/ui/DesktopNavigation.module.scss";
+import { push as SidebarPushPanel } from "react-burger-menu";
+import { GoogleTranslate } from "components/ui/GoogleTranslate";
+import {
+  StrapiModel,
+  extractLogoFromNavigationResponse,
+  extractNavigationMenusFromNavigationResponse,
+  ExtractedNavigationMenusFromNavigationResponse,
+} from "../../models/Strapi";
+import { PopUpMessage, PopupMessageProp } from "./PopUpMessage";
+import { Router } from "../../Router";
+import { useNavigationData } from "../../hooks/StrapiAPI";
+import { OUTER_CONTAINER_ID } from "../../App";
+
+const PAGE_WRAP_ID = "page-wrap";
+const BURGER_STYLES = {
+  bmBurgerButton: {
+    display: "none",
+  },
+  bmCrossButton: {
+    display: "none",
+  },
+  bmMenu: {
+    padding: "0",
+    borderLeft: "1px solid #f4f4f4",
+    background: "white",
+  },
+  bmOverlay: {
+    display: "none",
+  },
+};
+
+export const Navigation = () => {
+  const { data: navigationResponse, error, isLoading } = useNavigationData();
+  const [mobileNavigationSidebarIsOpen, setMobileNavigationSidebarIsOpen] =
+    useState(false);
+  const [whichActiveMobileSubMenu, setActiveMobileSubMenu] = useState("");
+  const toggleMobileNav = () =>
+    setMobileNavigationSidebarIsOpen((prev) => !prev);
+  const [whichActiveDesktopSubMenu, setsetActiveDesktopSubMenu] = useState("");
+  const [popUpMessage, setPopUpMessage] = useState<PopupMessageProp>({
+    message: "",
+    visible: false,
+    type: "success",
+  });
+
+  const logoData = extractLogoFromNavigationResponse(navigationResponse);
+  const menuData =
+    extractNavigationMenusFromNavigationResponse(navigationResponse);
+  const mobileSubMenuIsActive = !!whichActiveMobileSubMenu;
+  const desktopSubMenuIsActive = !!whichActiveDesktopSubMenu;
+
+  const pushPanelIconDisplay = () => {
+    if (mobileNavigationSidebarIsOpen && mobileSubMenuIsActive) {
+      return "fa-arrow-left";
+    }
+    if (mobileNavigationSidebarIsOpen) {
+      return "fa-xmark";
+    }
+
+    return ` fa-bars`;
+  };
+
+  const handleActivatePushPanelClick = () => {
+    if (mobileSubMenuIsActive) {
+      setActiveMobileSubMenu("");
+    } else {
+      toggleMobileNav();
+    }
+  };
+
+  const togglesetActiveDesktopSubMenu = (next: string) => {
+    if (desktopSubMenuIsActive && whichActiveDesktopSubMenu === next) {
+      setsetActiveDesktopSubMenu("");
+    } else {
+      setsetActiveDesktopSubMenu(next);
+    }
+  };
+
+  // TODO: What do we want here?
+  if (error || menuData === null) {
+    return <span>ERROR</span>;
+  }
+
+  // TODO: What do we want here?
+  if (isLoading) {
+    return <span>is loading...</span>;
+  }
```

Comment:
As I noted in one my of PRs, my preference would be to not indicate that its loading.